python/ray/dashboard/subprocesses/handle.py

```diff
@@ -204,206 +190,121 @@ async def _do_periodic_health_check(self):
-        exception: Exception,
-        response_fut: Awaitable[aiohttp.web.StreamResponse],
-    ) -> None:
         """
-        When the async iterator in the module raises an error, we need to propagate it
-        to the client and close the stream. However, we already sent a 200 OK to the
-        client and can't change that to a 500. We can't just raise an exception here to
-        aiohttp because that causes it to abruptly close the connection and the client
-        will raise a ClientPayloadError(TransferEncodingError).
-
-        Instead, we write exception to the stream and close the stream.
+        Proxy handler for WebSocket API
+        It establishes a WebSocket connection with the client and simultaneously connects
+        to the backend server's WebSocket endpoint. Messages are forwarded in both directions.
         """
-        try:
-            response = await asyncio.wrap_future(prev_fut)
-            await response.write(str(exception).encode())
-            await response.write_eof()
-            response_fut.set_result(response)
-        except Exception as e:
-            response_fut.set_exception(e)
-
-    def handle_parent_bound_message(self, message: ParentBoundMessage):
-        """Handles a message from the parent bound queue. This function must run on a
-        dedicated thread, called by dispatch_parent_bound_messages."""
-        loop = self.loop
-        if isinstance(message, UnaryResponseMessage):
-            active_request = self.active_requests.pop_or_raise(message.request_id)
-            # set_result is not thread safe.
-            loop.call_soon_threadsafe(
-                active_request.response_fut.set_result,
-                aiohttp.web.Response(
-                    status=message.status,
-                    body=message.body,
-                ),
-            )
-        elif isinstance(message, StreamResponseStartMessage):
-            active_request = self.active_requests.get_or_raise(message.request_id)
-            assert active_request.stream_response is None
-            # This assignment is thread safe, because a next read will come from another
-            # handle_parent_bound_message call for a Stream.*Message, which will run on
-            # the same thread and hence will happen-after this assignment.
-            active_request.stream_response = asyncio.run_coroutine_threadsafe(
-                SubprocessModuleHandle.handle_stream_response_start(
-                    active_request.request, message.body
-                ),
-                loop,
+        ws_from_client = aiohttp.web.WebSocketResponse()
+        await ws_from_client.prepare(request)
+
+        url = f"http://localhost{request.path_qs}"
+
+        async with self.session.ws_connect(
+            url, headers=filter_headers(request.headers)
+        ) as ws_to_backend:
+
+            async def forward(ws_source, ws_target):
+                async for msg in ws_source:
```

Comment:
so this is equivalent to

```
while True:
            msg = await ws.receive()

            if msg.type == aiohttp.WSMsgType.TEXT:
                yield msg.data
            elif msg.type == aiohttp.WSMsgType.CLOSED:
                break
            elif msg.type == aiohttp.WSMsgType.ERROR:
                pass
```

?

Reply:
No. It breaks the loop for message type in `(WSMsgType.CLOSE, WSMsgType.CLOSING, WSMsgType.CLOSED)`. See the source code:

https://github.com/aio-libs/aiohttp/blob/4399a6c614947bcb8566201c4ecc5133e7dfa4d6/aiohttp/web_ws.py#L609-L621